core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -346,7 +372,11 @@ private[zio] final class FiberContext[E, A](
    * Forks an `IO` with the specified failure handler.
    */
   final def fork[E, A](io: IO[E, A]): FiberContext[E, A] = {
-    val context = new FiberContext[E, A](platform, environment.peek())
+    val childFiberRefLocals: FiberRefLocals = platform.newWeakHashMap()
+    fiberRefLocals.synchronized {
+      childFiberRefLocals.putAll(fiberRefLocals)
```

Comment:
After seeing all these, I think it's a good idea to put the synchronization into `newWeakMap` in `Platform`.